tracim/tracim/lib/email_fetcher.py

```diff
@@ -0,0 +1,296 @@
+            -> None:
+        """
+        Fetch mail from a mailbox folder through IMAP and add their content to
+        Tracim through http according to mail Headers.
+        Fetch is regular.
+        :param host: imap server hostname
+        :param port: imap connection port
+        :param user: user login of mailbox
+        :param password: user password of mailbox
+        :param folder: mail folder where new mail are fetched
+        :param delay: seconds to wait before fetching new mail again
+        :param endpoint: tracim http endpoint where decoded mail are send.
+        :param token: token to authenticate http connexion
+        """
+        self._connection = None
+        self._mails = []
+        self.host = host
+        self.port = port
+        self.user = user
+        self.password = password
+        self.folder = folder
+        self.delay = delay
+        self.endpoint = endpoint
+        self.token = token
+
+        self._is_active = True
+
+    def run(self) -> None:
+        while self._is_active:
+            time.sleep(self.delay)
+            self._connect()
+            self._fetch()
+            self._notify_tracim()
+            self._disconnect()
+
+    def stop(self) -> None:
+        self._is_active = False
+
+    def _connect(self) -> None:
+        # FIXME - G.M - 2017-11-15 Verify connection/disconnection
+        # Are old connexion properly close this way ?
+        if self._connection:
+            self._disconnect()
+        # TODO - G.M - 2017-11-15 Support unencrypted connection ?
+        # TODO - G.M - 2017-11-15 Support for keyfile,certfile ?
+        self._connection = imaplib.IMAP4_SSL(self.host, self.port)
+        try:
+            self._connection.login(self.user, self.password)
+        except Exception as e:
+            log = 'IMAP login error: {}'
+            logger.debug(self, log.format(e.__str__()))
+
+    def _disconnect(self) -> None:
+        if self._connection:
+            self._connection.close()
+            self._connection.logout()
+            self._connection = None
+
+    def _fetch(self) -> None:
+        """
+        Get news message from mailbox
+        """
+
+        # select mailbox
+        rv, data = self._connection.select(self.folder)
+        if rv == 'OK':
+            # get mails
+            # FIXME - G.M -  2017-11-15 Which files to select as new file ?
+            # Unseen file or All file from a directory (old one should be moved/
+            # deleted from mailbox during this process) ?
+            rv, data = self._connection.search(None, "(UNSEEN)")
+            if rv == 'OK':
+                # get mail content
+                for num in data[0].split():
+                    rv, data = self._connection.fetch(num, '(RFC822)')
+                    if rv == 'OK':
+                        msg = message_from_bytes(data[0][1])
+                        decodedmsg = DecodedMail(msg)
+                        self._mails.append(decodedmsg)
```

Comment:
instead of storing mails in self._mails, why not returning them at the end of _fetch() call ?